Misc. trivial typo fixes

luzpaz opened the pull request:
Typos found via `codespell`. Typos discovered are mainly trivial source comment typos. There is an occasional important typo fix.

luzpaz commented on sympy/combinatorics/perm_groups.py:
```diff
@@ -1958,7 +1958,7 @@ def _number_blocks(blocks):
                 for i, r in enumerate(rep_blocks):
                     if len(r) > len(rep) and rep.issubset(r):
-                        # i-th block sytem is not minimal
+                        # i-th block system is not minimal
```
I'm not certain this fix is correct. Please review

luzpaz commented on sympy/combinatorics/partitions.py:
```diff
@@ -318,7 +318,7 @@ def __new__(cls, partition, integer=None):
         IntegerPartition(5, (2, 1, 1, 1))
 
-        If the value that the partion should sum to is given first, a check
+        If the value that the portion should sum to is given first, a check
```
I'm not sure this fix is correct, please review.

asmeurer replied:
I believe these should be "partition".

luzpaz commented on sympy/utilities/iterables.py:
```diff
@@ -1068,7 +1068,7 @@ def multiset_permutations(m, size=None, g=None):
 def _partition(seq, vector, m=None):
     """
-    Return the partion of seq as specified by the partition vector.
+    Return the portion of seq as specified by the partition vector.
```
I'm not sure this fix is correction either, please review

luzpaz commented on sympy/utilities/enumerative.py:
```diff
@@ -36,7 +36,7 @@
 equal, compare the second parts, and so on.
 
-In this ordering, the greatest partion of a given multiset has only
+In this ordering, the greatest portion of a given multiset has only
```
I'm not sure this fix is correct either, please review

asmeurer commented on sympy/core/multidimensional.py:
```diff
@@ -23,7 +23,7 @@ def apply_on_element(f, args, kwargs, n):
         is_arg = False
 
-    # Define reduced function that is only dependend of the specified argument.
+    # Define reduced function that is only dependent of the specified argument.
```
Should be "dependent on".

asmeurer commented on sympy/printing/fcode.py:
```diff
@@ -409,7 +409,7 @@ def split_pos_code(line, endpos):
                     return endpos
             return pos
-        # split line by line and add the splitted lines to result
+        # split line by line and add the split lines to result
```
Is this change correct?

luzpaz replied:
@asmeurer https://www.quora.com/Why-is-the-word-splitted-not-grammatically-correct

asmeurer commented:
I went through all the changes and they all look good except for those comments.

Commit:
Misc. trivial typo fixes

Typos found via `codespell`. Typos discovered are mainly trivial source comment typos. There is an occasional important typo fix.

luzpaz commented:
@asmeurer made the appropriate changes per your review. Repushed PR. Thanks you.

gxyd reviewed (approved):
I am generally not inclined towards PR's like these, which don't focus on changes to one section of code. But as Aaron has given +1. I'm merging these.

Also we tend to be a little negative towards force push.

Thanks for your contribution.

asmeurer commented:
Spelling fixes are fine. Unlike code style fixes, spelling is well agreed upon, and a spelling change is unlikely to have unintended consequences.
